### PR TITLE
Handle todos

### PR DESCRIPTION
**Pull request by andreasjansson**
Handle various TODOs, broken up into individual commits.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -28,6 +28,17 @@ test:
...
+.PHONY: test-end-to-end
+test-end-to-end: install
+	cd end-to-end-test/ && $(MAKE)
+
+.PHONY: test-cog-library
+test-cog-library:
+	cd pkg/docker/ && pytest cog_test.py
+
+.PHONY: test-all
+test-all: test test-end-to-end test-cog-library
```
Comment (bfirsh) [inline Makefile:40]: `make test` should probably test everything, right?

File: pkg/docker/cog.py [hunk 2]
```diff
@@ -504,6 +504,11 @@ def validate_and_convert_inputs(
     return inputs
 
 
+def run_model(model, inputs, cleanup_functions):
+    result = model.run(**inputs)
+    if isinstance(result, Path):
+        cleanup_functions.append(lambda: shutil.rmtree(result))
+    return result
```
Comment (bfirsh) [inline pkg/docker/cog.py:509]: Not important, but the passing by reference took some mental gymnastics to figure out what was going on. Maybe a clearer implementation would be to return `result, cleanup_function` where the latter is either a function or `None`.

File: pkg/docker/cog.py [hunk 3]
```diff
@@ -158,7 +158,7 @@ def handle_request():
                         )
                     except InputValidationError as e:
                         return jsonify({"error": str(e)})
-                    results.append(self.model.run(**instance))
+                    results.append(run_model(self.model, instance, cleanup_functions))
```
Comment (bfirsh) [inline pkg/docker/cog.py:161]: Looks like `cleanup_functions` are never called here.

File: pkg/docker/cog.py [hunk 4]
```diff
@@ -506,7 +506,7 @@ def validate_and_convert_inputs(
 def run_model(model, inputs, cleanup_functions):
     result = model.run(**inputs)
     if isinstance(result, Path):
-        cleanup_functions.append(lambda: shutil.rmtree(result))
+        cleanup_functions.append(lambda: result.unlink())
```
Comment (bfirsh) [inline pkg/docker/cog.py:508]: Can be squashed.

**Review by bfirsh — APPROVED** [review]
Great! Some minor things, but nothing blocking merging.

This is based on #82 which I'm not sure is intentional or not. Happy to include that if you want to get it in and we can iterate, otherwise might want to base this on `main` if we want to review that separately.

File: pkg/docker/cog.py (hunk 4 above)
Comment (andreasjansson) [reply pkg/docker/cog.py]: 👍

File: pkg/docker/cog.py (hunk 2 above)
Comment (andreasjansson) [reply pkg/docker/cog.py:519]: I added documentation to this function.

File: Makefile (hunk 1 above)
Comment (andreasjansson) [reply Makefile]: Yep, done. The old `make test` is now `make test-go`.

**Pull request merged**